main.go

```diff
@@ -46,6 +46,9 @@ func main() {
 
 	for {
 		if snapshotter.TokenExpiration.Before(time.Now()) {
+			if c.VaultAuthMethod == "k8s" {
```

Comment:
Won't this be overridden by the `snapshotter.SetClientTokenFromAppRole`?

Might want to convert this into a switch of some sort:

```
switch c.VaultAuthMethod {
case "k8s":
  snapshotter.SetClientTokenFromK8sAuth(c)
default:
  snapshotter.SetClientTokenFromAppRole(c)
}
```

Reply:
no that's alright, updated  this :+1: .